src/Avalonia.Controls/TextBlock.cs

```diff
@@ -364,14 +364,21 @@ protected virtual FormattedText CreateFormattedText(Size constraint)
         /// Invalidates <see cref="FormattedText"/>.
         /// </summary>
         protected void InvalidateFormattedText()
+        {
+            ClearFormattedText();
+            InvalidateMeasure();
+        }
+
+        /// <summary>
+        /// Invalidates <see cref="FormattedText"/>.
+        /// </summary>
```

Comment:
I'm not so keen on having both `InvalidateFormattedText`  and `ClearFormattedText`. Could we just have `InvalidateFormattedText` and then explicitly call `InvalidateMeasure` when needed?

Reply:
Yes, thats possible. I have kept those 2 function gets used in same call in order to not to break those who are already using that function. To implement your proposal we also need to update thoose places who are usnig InvalidateFormattedText to also call InvalidateMeasure. 
Then we dont need a function who do both

Reply:
Let me know if you would like to do that or not, if so I update those who are right now using InvalidateFormattedText

Reply:
I'm fine with changing the behavior of `InvalidateFormattedText` at this point - we're still in beta, we can do stuff like that ;) So yeah go ahead and make the change.